src/ray/raylet/worker_pool.cc

```diff
@@ -1443,7 +1443,9 @@ inline bool WorkerPool::IsIOWorkerType(const rpc::WorkerType &worker_type) const
 }
 
 const std::vector<std::shared_ptr<WorkerInterface>> WorkerPool::GetAllRegisteredWorkers(
-    bool filter_dead_workers, bool filter_io_workers) const {
+    bool filter_dead_workers,
+    bool filter_io_workers,
+    bool filter_non_retriable_workers) const {
```

Comment:
Nit, but I think it'd be a bit better to not put this logic inside the WorkerPool. Then we can keep all the logic in a separate memory monitor class and unit-test it there.